src/Core/Components/List/FluentAutocomplete.razor.cs

```diff
@@ -104,6 +106,7 @@ public override bool Multiple
 
     /// <summary>
     /// Gets or sets the number of maximum options (items) returned by <see cref="OnOptionsSearch"/>.
+    /// If the value is smaller than 0, all search results are shown.
```

Comment:
I don't like this 'magic parameter' where you use a negative number to force a certain behavior. Think it is better to introduce a separate (new) parameter for that.

Reply:
Would it not be slower? Component needs to be invoked, do some calculations and then come to the conclusion nothing needs to be done.

Reply:
To validate, but I don't think so. I think that only `LoadFilteredItemsAsync` will be call to get items.

Reply:
I introduced the _Virtualize_ Parameter because FluentDataGrid also has one. And in the comment of the FluentDataGrid's parameter it is [mentioned](https://github.com/microsoft/fluentui-blazor/blob/dev/src/Core/Components/DataGrid/FluentDataGrid.razor.cs#L43) that:
"Generally it's preferable not to use <see cref="Virtualize"/> if the amount of data being rendered is small or if you are using pagination."

Therefore, I thought that it is better to let the user decide because there may be performance drawbacks if using Virtualization with only a few items.

Reply:
Yeah, I like that better too. Make it a consious choice made by the developer. I propose to not make it always use `Virlualize` (ie leave it as you have it now).

Remark about the `magic parameter` still standss though...

Reply:
Which solution is prefered by you, regarding the MaximumOptionsSearch parameter
1. New Parameter for disabling this option
2. Remove -1 logic and just use e.g. int.Max

I think both are better than the actual implemented behaviour in the PR because then we do not have a breaking change.

Reply:
Can't we just leave as is an in case of `Virtualize` being used ignore the `MaximumOptionSearch` value?

Reply:
Regarding always Virtualize or not. If the Virtualize parameter is introduced, we can be sure that no different behaviour is introduced for projects that use the AutoComplete component. So everyone have to switch manually if needed.

Different behaviour: This can for example be the case if items with a different height are in place. The Virtualize component sets the item height to a fixed height of 50px by default. I toke over this default value to the AutoComplete ItemsHeight parameter.

Reply:
> Regarding always Virtualize or not. If the Virtualize parameter is introduced, we can be sure that no different behaviour is introduced for projects that use the AutoComplete component. So everyone have to switch manually if needed.

Agreed. Lets not make it do Virtualize always

> Different behaviour: This can for example be the case if items with a different height are in place. The Virtualize component sets the item height to a fixed height of 50px by default. I took over this default value to the AutoComplete ItemsHeight parameter.

Default is good, I think. Believe it is at 32 for the DataGrid.

Reply:
Martin, I'm preparing 4.5 release right now. Can we get this finished and merged in today?

Reply:
You mean default stays at 9 and dev can set it to int.Max if they want everything? If os, yes that seems like a good approach.

Reply:
Exactly.